packages/@aws-cdk/cx-api/lib/assembly/cloudformation-stack.ts

```diff
@@ -0,0 +1,57 @@
+import { Artifact, AwsCloudFormationStackProperties } from '../artifacts';
+import { ASSET_METADATA, AssetMetadataEntry } from '../metadata/assets';
+import { CloudArtifact } from './cloud-artifact';
+import { CloudAssembly } from './cloud-assembly';
+import { ICloudFormationStackArtifact } from './cloud-assembly-api';
+
+export class CloudFormationStackArtifact extends CloudArtifact implements ICloudFormationStackArtifact {
+  public readonly template: any;
+  public readonly originalName: string;
+  public readonly logicalIdToPathMap: { [logicalId: string]: string };
+  public readonly assets: AssetMetadataEntry[];
+
+  public name: string;
+
+  constructor(assembly: CloudAssembly, name: string, artifact: Artifact) {
+    super(assembly, name, artifact);
+
+    if (!artifact.properties) {
+      throw new Error(`Invalid CloudFormation stack artifact. Missing properties`);
+    }
+
+    const properties = artifact.properties as AwsCloudFormationStackProperties;
+    this.template = this.assembly.readJson(properties.templateFile);
+    this.originalName = name;
+    this.name = this.originalName;
+    this.logicalIdToPathMap = this.buildLogicalToPathMap();
+    this.assets = this.buildAssets();
+  }
+
+  private buildAssets() {
```

Comment:
This sort of suggested to me that the actual build process was happening here - but this really just makes an array... Could call `collectAssets` instead.